File: src/main/java/com/interviewplannerapp/service/MeetingNoteService.java
package com.interviewplannerapp.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.interviewplannerapp.domain.MeetingNote;
import com.interviewplannerapp.dto.MeetingNoteDTO;
import com.interviewplannerapp.dto.MeetingNoteSearchDTO;
import com.interviewplannerapp.dto.MeetingNotePageDTO;
import com.interviewplannerapp.dto.MeetingNoteConvertCriteriaDTO;
import com.interviewplannerapp.service.GenericService;
import com.interviewplannerapp.dto.common.RequestDTO;
import com.interviewplannerapp.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface MeetingNoteService extends GenericService<MeetingNote, Integer> {

	List<MeetingNote> findAll();

	ResultDTO addMeetingNote(MeetingNoteDTO meetingNoteDTO, RequestDTO requestDTO);

	ResultDTO updateMeetingNote(MeetingNoteDTO meetingNoteDTO, RequestDTO requestDTO);

    Page<MeetingNote> getAllMeetingNotes(Pageable pageable);

    Page<MeetingNote> getAllMeetingNotes(Specification<MeetingNote> spec, Pageable pageable);

	ResponseEntity<MeetingNotePageDTO> getMeetingNotes(MeetingNoteSearchDTO meetingNoteSearchDTO);
	
	List<MeetingNoteDTO> convertMeetingNotesToMeetingNoteDTOs(List<MeetingNote> meetingNotes, MeetingNoteConvertCriteriaDTO convertCriteria);

	MeetingNoteDTO getMeetingNoteDTOById(Integer meetingNoteId);







}

File: interviewplannerapp_ui/src/services/MeetingNoteService.js
import http from "../http-common"; 

class MeetingNoteService {
  getAllMeetingNotes(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/meetingNote/meetingNotes`, searchDTO);
  }

  get(meetingNoteId) {
    return this.getRequest(`/meetingNote/${meetingNoteId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/meetingNote?field=${matchData}`, null);
  }

  addMeetingNote(data) {
    return http.post("/meetingNote/addMeetingNote", data);
  }

  update(data) {
  	return http.post("/meetingNote/updateMeetingNote", data);
  }
  
  uploadImage(data,meetingNoteId) {
  	return http.postForm("/meetingNote/uploadImage/"+meetingNoteId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new MeetingNoteService();
